spyderlib/plugins/editor.py

```diff
@@ -1894,7 +1899,16 @@ def save_as(self):
             if CONF.get('workingdir', 'editor/save/auto_set_to_basedir'):
                 self.open_dir.emit(osp.dirname(fname))
             self.__add_recent_file(fname)
-    
+
+    @Slot()
+    def save_copy_as(self):
+        """Save *copy as* the currently edited file"""
+        editorstack = self.get_current_editorstack()
+        if editorstack.save_copy_as():
+            fname = editorstack.get_current_filename()
+            if CONF.get('workingdir', 'editor/save/auto_set_to_basedir'):
+                self.emit(SIGNAL("open_dir(QString)"), osp.dirname(fname))

```

Comment:
We don't support the old style of signals and slots anymore because it's incompatible with PyQt5. You need to change this instead to:

    self.open_dir.emit(osp.dirname(fname))